lib/value.ml

```diff
@@ -392,16 +407,22 @@ let of_yaml_record_to_expr ~loc fields =
           [%expr Error (`Msg (x ^ Yaml.to_string_exn y))];
       ]
   in
-  let option_to_none t =
-    match t.pld_type with
-    | [%type: [%t? _] option] -> [%expr Ok None]
-    | _ ->
+  let option_to_none t = 
+    match Attribute.get Attrs.default t with
+    | None -> [%expr Ok None]
+    | Some default -> 
+      (match t.pld_type with
+      | [%type: [%t? _] option] -> 
+        (* default *)
+        let default = [%expr ([%e default] : [%t t.pld_type])] in
+        [%expr Ok [%e default]]
+      | _ ->
```

Comment:
Currently `ppx_deriving_yaml` supports one default value which is supplying `None` whenever we see an optional field in a record. Your PR adds a very useful feature where we can supply an attribute to tell the ppx what value to use. That is, in the event of missing data in the yaml that we are trying to convert to an OCaml value, we should supply the default value and not return an `Error`. 

The logic for this feature is:
 1. Did the user supply a `[@default value]` for this field? If yes go to (2) if no go to (3).
 2. Return the default value as `[%expr Ok [%e default]]`. Done.
 3. We have no default value, but is the type something we could supply one for like we currently have for option types? If it is we can supply that, otherwise we return an `Error`.

Your code is very close to this but subtly incorrect. 

```ocaml
 match Attribute.get Attrs.default t with
    | None -> [%expr Ok None]
```

Here you do step (1) by checking if we have a default attribute. If we don't (the `None` case) then you have returned `Ok None` but this isn't quite right since we don't know at this point whether the type we are generating code for (i.e. `t.pld_type`) is in fact some `'a option`. So you will need to check this like the old code did with `match t.pld_type with | [%type: [%t? _] option] -> [%expr Ok None]`.

```ocaml
| Some default -> 
      (match t.pld_type with
      | [%type: [%t? _] option] -> 
        (* default *)
        let default = [%expr ([%e default] : [%t t.pld_type])] in
        [%expr Ok [%e default]]
      | _ ->
```

In your code for where we do have a default value, we no longer need to check what kind of type `t.pld_type` is since we are trusting the user to supply us with a value of the right type. So in fact this can be simplified to:

```ocaml
| Some default -> 
  let default = [%expr ([%e default] : [%t t.pld_type])] in
  [%expr Ok [%e default]]
```